src/AstGen.zig

```diff
@@ -8398,6 +8403,22 @@ fn rvalue(
     rl: ResultLoc,
     result: Zir.Inst.Ref,
     src_node: ast.Node.Index,
+) InnerError!Zir.Inst.Ref {
+    if (gz.refIsNoReturn(result)) {
+        return gz.astgen.failNode(src_node, "unreachable code", .{});
+    }
+    return rvalueMaybeUnreachable(gz, rl, result, src_node);
+}
+
+/// Applies `rl` semantics to `inst`. Expressions which do not do their own handling of
+/// result locations must call this function on their result.
+/// As an example, if the `ResultLoc` is `ptr`, it will write the result to the pointer.
+/// If the `ResultLoc` is `ty`, it will coerce the result to the type.
+fn rvalueMaybeUnreachable(
```

Comment:
Why is this split needed? It should be documented why this variant needs to exist and when to use it in the doc comment instead of duplicating the information already in the doc comment of `rvalue()`

Reply:
https://github.com/ziglang/zig/pull/9645/files/b0f73a6847e12373ec938c0570e3b9fb93c8df28#diff-3ed5c754ecaac0e7ed512aafc83f6cb0acb474fc267eff3efa775e1afdad159dR7529-R7534